pkg/primitives/vm/vm.go

```diff
@@ -325,6 +326,12 @@ func (p *Manager) Deprovision(ctx context.Context, wl *gridtypes.WorkloadWithID)
 		log.Error().Err(err).Str("name", volName).Msg("failed to delete rootfs volume")
 	}
 
+	cidataPath := filepath.Join("var", "cache", "modules", "vmd", "cloud-init", wl.ID.String())
```

Comment:
While this will work, but it's completely out of place. The concept of separation of concerns dictates that a single module/object should not know about the details implementations of another module. 

It would be logical that the module that actually creates the cidata file is the one that also deletes it. no?

Otherwise the code logic will be scattered all over the place and will be completely unmanageable.

Reply:
You can see already that you had to define a path that is completely static. what if the modules that creates the cidata now decided to change this path, is this still gonna work?